### PR TITLE
Fix URDF and LPCL for SC

### PR DESCRIPTION
**Pull request by SamerKhshiboun**
- handle special (SC) streams with ROS system
- add parameters (SPreset x, y, z, raw, pitch, yaw) to URDF, if user want to use hard coded ROS Transformations for simulation (rviz2, gazebo, etc..)

### CODE REVIEW

**Review by Nir-Az — APPROVED** [review]
LGTM, waiting for @benlev confirmation

**Comment by SamerKhshiboun** [comment]
There is something still missing related to the RPY (if given) of the safety preset.
I will fix/test that tomorrow.

File: realsense2_camera/src/tfs.cpp [hunk 1]
```diff
@@ -168,12 +168,36 @@ void BaseRealSenseNode::calcAndAppendTransformMsgs(const rs2::stream_profile& pr
...
+        // All other streams are Optical streams, so we use the "Q" quaternion
+        // Also here, the translation vector is in the Optical CS, and we convert it to ROS CS
+        // inside append_static_tf_msg
+        append_static_tf_msg(transform_ts_, trans, Q, _base_frame_id, FRAME_ID(sip));
+    }
 
-    // Transform stream frame to stream optical frame and publish it
+    // Transform stream frame from ROS CS to optical CS and publish it
+    // We are using zero translation vector here, since no translation between frame and optical_frame, but only rotation
     append_static_tf_msg(transform_ts_, zero_trans, quaternion_optical, FRAME_ID(sip), OPTICAL_FRAME_ID(sip));
```
Comment (Nir-Az) [inline realsense2_camera/src/tfs.cpp:216]: We need 3 calls to ` append_static_tf_msg`?
if ...
   ` append_static_tf_msg`
else
   ' append_static_tf_msg'

' append_static_tf_msg'
?

Comment (SamerKhshiboun) [reply realsense2_camera/src/tfs.cpp:216]: We expose two frames for each sensor
ROS Frame
Optical Frame

The first if/else, includes a append_static_tf_msg that is exposing the ROS_FRAME
the second append_static_tf_msg is exposing the OPTICAL_FRAME (is just a rotation of the ROS_FRAME, with zero translation)

**Comment by benlev** [comment]
Tried this in my environment - seems to work well.

**Review by benlev — APPROVED** [review]

**Pull request merged**